src/ucp/wireup/wireup_cm.c

```diff
@@ -617,6 +643,8 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
               ucp_ep, ucp_ep->flags, ucp_ep->cfg_index,
               ucs_status_string(status));
 
+    UCP_CM_CB_CHECK_UCP_EP(ucp_ep, uct_cm_ep, return);
```

Comment:
why is it ok to check ep flags w/o async lock?

Reply:
added check that async lock is blocked when checking this flag